### ISSUE TITLE
differential expression crashes on large dataset

### ISSUE THREAD

**Issue by neuromusic** [chanzuckerberg/cellxgene#718]
I have a 1M+ cell dataset. `.X` is a sparse uint16 array. Computer is a 16GB MBP.

I select ~170k cells and assign them to cell set 1, then select the remaining cells and assign them to cell set 2. Click "compute differential expression" and the wheel spins for a few minutes while my fan kicks on. Then the cellxgene server crashes, printing `Killed: 9` to the console.

```
[cellxgene] Launching! Opening your browser to http://127.0.0.1:5005 now.
[cellxgene] Type CTRL-C at any time to exit.
Killed: 9
```

**Comment by neuromusic** [comment]
same issue w/ sparse float32 values

**Comment by neuromusic** [comment]
ok, it seems likely this issue is due to the matrix being stored in CSR format 🤦‍♂️ 

going to try CSC and close this issue if that's the cause

**Comment by bkmartinjr** [comment]
This is most likely the same root issue as #718.  Tagged for 0.10.

**Comment by bkmartinjr** [comment]
This turns out to be more closely related to #739, as the root issue is a file where the working set is too large for main memory.  At a minimum, we should implement some of the guardrails discussed in that issue.

**Comment by bkmartinjr** [comment]
Digging into this, it looks like the root cause is that your host ran out of VM.  I propose that we close this as "will not fix", but partly address it with #739.   Please comment.
